### DISCUSSION TITLE
Problem with  dpgen run "AttributeError: 'str' object has no attribute 'from_str'"

### DISCUSSION THREAD

**Discussion by Nancy877** [deepmodeling/dpgen#1282] [Q&A]
Hello teacher,
When I use dpgen run, I meet the following errors:
INFO:dpgen:-------------------------iter.000000 task 05--------------------------
INFO:dpgen:-------------------------iter.000000 task 06--------------------------
INFO:dpgen:system 000 candidate :     56 in     93  60.22 %
INFO:dpgen:system 000 failed    :     35 in     93  37.63 %
INFO:dpgen:system 000 accurate  :      2 in     93   2.15 %
INFO:dpgen:system 000 accurate_ratio:   0.0215    thresholds: 1.0000 and 1.0000   eff. task min and max   -1    5   number of fp tasks:      5
Traceback (most recent call last):
  File "/opt/software/deepmd-kit/bin/dpgen", line 8, in <module>
    sys.exit(main())
  File "/opt/software/deepmd-kit/lib/python3.10/site-packages/dpgen/main.py", line 233, in main
    args.func(args)
  File "/opt/software/deepmd-kit/lib/python3.10/site-packages/dpgen/generator/run.py", line 5109, in gen_run
    run_iter(args.PARAM, args.MACHINE)
  File "/opt/software/deepmd-kit/lib/python3.10/site-packages/dpgen/generator/run.py", line 4458, in run_iter
    make_fp(ii, jdata, mdata)
  File "/opt/software/deepmd-kit/lib/python3.10/site-packages/dpgen/generator/run.py", line 3516, in make_fp
    make_fp_vasp(iter_index, jdata)
  File "/opt/software/deepmd-kit/lib/python3.10/site-packages/dpgen/generator/run.py", line 3157, in make_fp_vasp
    make_fp_vasp_kp(iter_index, jdata)
  File "/opt/software/deepmd-kit/lib/python3.10/site-packages/dpgen/generator/run.py", line 2933, in make_fp_vasp_kp
    standard_incar = incar_upper(Incar.from_string(incar))
  File "/opt/software/deepmd-kit/lib/python3.10/site-packages/numpy/lib/utils.py", line 97, in newfunc
    return func(*args, **kwds)
  File "/opt/software/deepmd-kit/lib/python3.10/site-packages/pymatgen/io/vasp/inputs.py", line 766, in from_string
    return cls.from_str(*args, **kwargs)
AttributeError: 'str' object has no attribute 'from_str'

The installed software versions are as follows:
DeePMD-kit v2.2.2
dpgen Version: 0.11.1
Path:    /opt/software/deepmd-kit/lib/python3.10/site-packages/dpgen
Dependency
------------
     numpy     1.22.3   /opt/software/deepmd-kit/lib/python3.10/site-packages/numpy
    dpdata     0.2.15   /opt/software/deepmd-kit/lib/python3.10/site-packages/dpdata
  pymatgen            unknown version or path
     monty  2022.1.19   /opt/software/deepmd-kit/lib/python3.10/site-packages/monty
       ase     3.22.1   /opt/software/deepmd-kit/lib/python3.10/site-packages/ase
  paramiko      3.2.0   /opt/software/deepmd-kit/lib/python3.10/site-packages/paramiko
 custodian   2023.6.5   /opt/software/deepmd-kit/lib/python3.10/site-packages/custodian
DeePMD-kit v2.2.2

I doubt  whether this caused by a mismatch in my Python version.

**Comment by njzjz** [comment]
The reason should be a breaking change in the pymatgen package. Could you use `pip list` to print its version? An old version of pymatgen should work.

**Reply by njzjz** [reply]
It seems to be a bug in https://github.com/materialsproject/pymatgen/pull/3158

**Comment by Nancy877** [comment]
Thanks for your reply！ I use "pip list", it shows:
pymatgen&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 2023.7.17
pymatgen-analysis-defects 2023.7.12



I have also read the relevant link you provided (materialsproject/pymatgen#3158）, but found that there are too many areas that need to be modified. Can I uninstall the new version of pymatgen and download a suitable old version? Could you recommend a version for me. Thanks very much!
（I previously installed dpgen 0.11.0,&nbsp; found no pymatgen in the pip list. I tried to lower the existing dpgen version to 0.11.0 and displayed the same error message）


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "deepmodeling/dpgen"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年7月22日(星期六) 下午4:09
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [deepmodeling/dpgen] Problem with  dpgen run "AttributeError: 'str' object has no attribute 'from_str'" (Discussion #1282)





 
The reason should be a breaking change in the pymatgen package. Could you use pip list to print its version? I guess an old version should work.

**Reply by Nancy877** [reply]
Thanks very much. I have changed the version with pymatgen=2023.5.8 and the software can run successfully！
